### PR TITLE
USDZExporter: Modulate diffuse map

### PR DESCRIPTION
**Pull request by kolodi**
Related issue: #21871

A material can have a color different from white. Color is rendered on top of existing map (diffuse).
Initially I was thinking to use the bias input for [Texture Reader](https://graphics.pixar.com/usd/docs/UsdPreviewSurface-Proposal.html#UsdPreviewSurfaceProposal-TextureReader#UsdPreviewSurfaceProposal-TextureReader), but unfortunately it does not worl, at least not in QuickLook.

Current solutions is more like a workaround and may be revisited some day.

### CODE REVIEW

File: build/three.js [hunk 1]
```diff
@@ -36127,3 +36127,4 @@
 	Object.defineProperty(exports, '__esModule', { value: true });
 
 })));
+//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidGhyZWUuanMiLCJzb3VyY2VzIjpbXSwic291cmNlc0NvbnRlbnQiOltdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiIn0=
```
Comment (Mugen87) [inline build/three.js:36130]: Do you mind removing the build files from the PR?

Comment (kolodi) [reply build/three.js]: Done

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Ended up refactoring the code quite a bit: https://github.com/mrdoob/three.js/commit/afb032b20e5116c921bc40d0095190ef2bcb82f8

**Comment by mrdoob** [comment]
This commit may shows better the implementation: https://github.com/mrdoob/three.js/commit/2ceaad92897fc10176fb69bd86a9f4f93ce0f3cf

**Comment by mrdoob** [comment]
Also, turns out I had broken `USDZExporter` here: https://github.com/mrdoob/three.js/commit/14fab092b66c9956f3f9f41d515a654a863dd3ca
Took me quite a few hours to figure out the fix: https://github.com/mrdoob/three.js/commit/a09a70474aa41f8445595e2232d7fc01e45c4a3d?w=1

**Comment by kolodi** [comment]
Very good. Next step - fix broken ao map ;-)

**Comment by kolodi** [comment]
See my other comments in 2ceaad9

**Comment by mrdoob** [comment]
> Very good. Next step - fix broken ao map ;-)

Yeah. Not sure how to fix that. As far as I understand USDZ only supports 1 uv channel.

**Comment by mrdoob** [comment]
By the way, there are a lot of tricks here:

https://github.com/google/usd_from_gltf#emulated-functionality-for-ar-quick-look

**Comment by kolodi** [comment]
> 
> 
> By the way, there are a lot of tricks here:
> 
> https://github.com/google/usd_from_gltf#emulated-functionality-for-ar-quick-look

Quite dated. For example, they ended up scaling textures to "simulate" tiling...

**Pull request merged**